include/scene.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef SCENE_HPP_
+#define SCENE_HPP_
+
+#include <boost/config.hpp>
+#include <memory>
+#include "enum_array.hpp"
+#include "noncopyable.hpp"
+#include "raylib-cpp.hpp"
+
+namespace meow {
+
+class SceneManager;
+
+class BOOST_SYMBOL_VISIBLE Scene : noncopyable {
+    friend class SceneManager;
+
+protected:
+    // NOLINTBEGIN cppcoreguidelines-non-private-member-variables-in-classes
+    raylib::Window *m_window = nullptr;
+    SceneManager *m_scene_manager = nullptr;
```

Comment:
I didn't really understand it is necessary to store a pointer to `SceneManager `. What is the purpose of this?

Reply:
to change scenes, for example, from lobby to game and back